src/include/wiredtiger.in

```diff
@@ -742,6 +742,10 @@ struct __wt_cursor {
      */
 	WT_ITEM lower_bound, upper_bound;
 
+	/* Format test program private callback and callback argument. */
+	void (*format_private)(int, void *);
+	void *format_private_arg;
```

Comment:
This is the change that makes using cursors unappealing. It's a change to the general public API of WiredTiger, even though it's undocumented. I'll think some more about whether the layering niceness warrants this intrusion into the public API.